subprojects/kotlin-dsl-tooling-builders/src/main/kotlin/org/gradle/kotlin/dsl/tooling/builders/KotlinDslScriptsModelBuilder.kt

```diff
@@ -247,6 +233,23 @@ data class KotlinDslScriptsParameter(
 )
 
 
+internal
+fun <T : Any> commonPrefixOf(lists: List<List<T>>): List<T> {
+    if (lists.isEmpty()) return emptyList()
+    if (lists.size == 1) return lists.first()
+    val maxIndex = lists.map { it.size }.min()!! - 1
+    if (maxIndex < 0) return emptyList()
+    val first = lists.first()
+    val others = lists.drop(1)
+    val common = ArrayList<T>(maxIndex)
+    for (idx in 0..maxIndex) {
+        if (others.all { it[idx] == first[idx] }) common += first[idx]
+        else break
+    }
+    return common
+}
```

Comment:
❌ May I suggest a more condensed version of the algorithm which doesn't create intermediate lists and seems a little easier to read?

```kotlin
internal
fun <T : Any> commonPrefixOf(lists: List<List<T>>): List<T> =
    lists.minBy { it.size }?.let { maxCommonPrefix ->
        maxCommonPrefix.indices.asSequence().takeWhile { index ->
            lists.all { list -> list[index] == maxCommonPrefix[index] }
        }.lastOrNull()?.let { maxCommonIndex ->
            maxCommonPrefix.take(maxCommonIndex + 1)
        }
    } ?: emptyList()
```

Reply:
Consider merging https://github.com/gradle/gradle/pull/11637